File: app/src/main/java/me/blog/korn123/easydiary/adapters/DiaryWeatherItemAdapter.java
package me.blog.korn123.easydiary.adapters;

import android.app.Activity;
import android.content.Context;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import me.blog.korn123.commons.constants.Constants;
import me.blog.korn123.commons.utils.CommonUtils;
import me.blog.korn123.commons.utils.EasyDiaryUtils;
import me.blog.korn123.commons.utils.FontUtils;
import me.blog.korn123.easydiary.R;
import me.blog.korn123.easydiary.activities.DiaryInsertActivity;
import me.blog.korn123.easydiary.extensions.ContextKt;

/**
 * Created by CHO HANJOONG on 2017-03-16.
 */

public class DiaryWeatherItemAdapter extends ArrayAdapter<String> {
    private final Context mContext;
    private final List<String> mList;
    private final int mLayoutResourceId;

    public DiaryWeatherItemAdapter(Context context, int layoutResourceId, List<String> list) {
        super(context, layoutResourceId, list);
        this.mContext = context;
        this.mList = list;
        this.mLayoutResourceId = layoutResourceId;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        return initRow(position, convertView, parent);
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        return initRow(position, convertView, parent);
    }

    private View initRow(int position, View convertView, ViewGroup parent) {
        View row = convertView;
        ViewHolder holder = null;
        if (row == null) {
            LayoutInflater inflater = ((Activity)this.mContext).getLayoutInflater();
            row = inflater.inflate(this.mLayoutResourceId, parent, false);
            holder = new ViewHolder();
            holder.textView1 = ((TextView)row.findViewById(R.id.text1));
            holder.imageView1 = ((ImageView) row.findViewById(R.id.imageView1));
            row.setTag(holder);
        } else {
            holder = (ViewHolder)row.getTag();
        }

        setFontsTypeface(holder);
        EasyDiaryUtils.initWeatherView(holder.imageView1, position, true);

        holder.textView1.setText(mList.get(position));
        if (position == 0) {
            holder.imageView1.setVisibility(View.GONE);
        } else {
            holder.imageView1.setVisibility(View.VISIBLE);
        }

        ContextKt.initTextSize(mContext, holder.textView1, mContext);
        return row;
    }

    private void setFontsTypeface(ViewHolder holder) {
        FontUtils.setFontsTypeface(mContext, mContext.getAssets(), null, holder.textView1);
    }

    private static class ViewHolder {
        TextView textView1;
        ImageView imageView1;
    }

}

File: app/src/main/java/me/blog/korn123/easydiary/adapters/DiaryWeatherItemAdapter.kt
package me.blog.korn123.easydiary.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import me.blog.korn123.commons.utils.FlavorUtils
import me.blog.korn123.commons.utils.FontUtils
import me.blog.korn123.easydiary.R
import me.blog.korn123.easydiary.extensions.config
import me.blog.korn123.easydiary.models.DiarySymbol

/**
 * Created by CHO HANJOONG on 2017-03-16.
 * Refactored code on 2019-12-26.
 *
 */

class DiaryWeatherItemAdapter(
        context: Context,
        private val layoutResourceId: Int,
        private val mList: List<DiarySymbol>
) : ArrayAdapter<DiarySymbol>(context, layoutResourceId, mList) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        return initRow(position, convertView, parent)
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View? {
        return initRow(position, convertView, parent)
    }

    private fun initRow(position: Int, convertView: View?, parent: ViewGroup): View {
        val itemView: View = convertView ?: LayoutInflater.from(parent.context).inflate(this.layoutResourceId, parent, false)

        when (itemView.tag is ViewHolder) {
            true -> itemView.tag as ViewHolder
            false -> {
                val viewHolder = ViewHolder(itemView.findViewById(R.id.text1), itemView.findViewById(R.id.imageView1), itemView.findViewById(R.id.item_holder))
                itemView.tag = viewHolder
                viewHolder
            }
        }.run {
            textView1.setTextColor(context.config.textColor)
            FontUtils.setFontsTypeface(context, context.assets, null, item_holder)
            FlavorUtils.initWeatherView(context, imageView1, mList[position].sequence, false)
            textView1.text = mList[position].description
            imageView1.visibility = if (mList[position].sequence == 0) View.GONE else View.VISIBLE
        }

        return itemView
    }

    private class ViewHolder (val textView1: TextView, val imageView1: ImageView, val item_holder: LinearLayout)
}
